cardano-api/src/Cardano/Api/DeserialiseAnyOf.hs

```diff
@@ -0,0 +1,272 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | Deserialisation of input that may be formatted/encoded in one of several specified
+-- ways.
+module Cardano.Api.DeserialiseAnyOf
+  ( InputFormat (..)
+  , InputDecodeError (..)
+  , deserialiseInput
+  , deserialiseInputAnyOf
+  , renderInputDecodeError
+
+  , readAndDeserialiseFromFile
+  , readAndDeserialiseFromFileAnyOf
+  , readAndDeserialiseFromTextEnvelopeFile
```

Comment:
The naming convention in the API for these sorts of things is `readFileBlahBlah`.

Reply:
Okay, I'll think of some better names. I think I was attempting to avoid clashing with `readFileTextEnvelope` and came up with a different naming scheme.